saleor/graphql/discount/tests/mutations/test_voucher_create.py

```diff
@@ -118,6 +118,55 @@ def test_create_voucher(
     assert len(codes) == 2
 
 
+def test_create_voucher_no_codes(
+    staff_api_client,
+    permission_manage_discounts,
+    product,
+    variant,
+    collection,
+    category,
+):
+    # given
+    start_date = timezone.now() - timedelta(days=365)
+    end_date = timezone.now() + timedelta(days=365)
+
+    name = "test voucher"
+    variables = {
+        "input": {
+            "name": name,
+            "type": VoucherTypeEnum.ENTIRE_ORDER.name,
+            "addCodes": [""],
+            "discountValueType": DiscountValueTypeEnum.FIXED.name,
+            "minCheckoutItemsQuantity": 10,
+            "startDate": start_date.isoformat(),
+            "endDate": end_date.isoformat(),
+            "applyOncePerOrder": True,
+            "applyOncePerCustomer": True,
+            "singleUse": True,
+            "usageLimit": 3,
```

Comment:
I see conflict here: `singleUse=True` while `usageLimit=3`. Are we sure it doesn't impact calculations? I.e `saleor.discount.utils.increase_voucher_usage` function will perform both `increase_voucher_code_usage_value()` and `deactivate_voucher_code()`.

Reply:
this combination will work but in this case we do not touch `increase_voucher_usage` because we just want to create Voucher and check if resolver on Voucher field `used` will work properly.